File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L1/07/Kasus7L1.ts
import { Scanner } from 'readline-sync';

function sumMajorDiagonal(m: number[][]): number {
  let sum = 0;
  for (let i = 0; i < m.length; i++) {
    sum += m[i][i];
  }
  return sum;
}

function main(): void {
  // Get input
  const input = new Scanner();
  console.log('Enter a 4 by 4 matrix row by row:');
  const m: number[][] = [];
  for (let i = 0; i < 4; i++) {
    m[i] = [];
    for (let j = 0; j < 4; j++) {
      m[i][j] = input.nextDouble();
    }
  }

  // Calculate and print the sum of the major diagonal
  console.log(`Sum of the elements in the major diagonal is ${sumMajorDiagonal(m)}`);
}

main();

File: datasets/IR-Plag/intellij-original-java-rest-kotlin/case-07/plagiarized/L1/07/Kasus7L1.kt
package IR

import java.util.*

/*
* To change this license header, choose License Headers in Project Properties.
* To change this template file, choose Tools | Templates
* and open the template in the editor.
*/

/**
 *
 * @author B15130F5DDB6B5F1622EF91DAC4C1AAE
 */
object Kasus7L1 {
    @JvmStatic
    fun main(args: Array<String>) {
        //minta input
        val input = Scanner(System.`in`)
        print("Enter a 4 by 4 matrix row by row: ")
        val m = Array(4) { DoubleArray(4) }
        //input the matrix
        for (i in 0..3) {
            for (j in 0..3) {
                m[i][j] = input.nextDouble()
            }
        }
        print("Sum of the elements in the major diagonal is " + sumMajorDiagonal(m))
    }

    fun sumMajorDiagonal(m: Array<DoubleArray>): Double {
        var sum = 0.0
        for (i in m.indices) {
            sum += m[i][i]
        }
        return sum
    }
}
